perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_member_resource_attribute_def_virt_isBanned.java

```diff
@@ -73,58 +81,148 @@ public Attribute getAttributeValue(PerunSessionImpl sess, Member member, Resourc
     }
 
 	@Override
-	public List<String> resolveVirtualAttributeValueChange(PerunSessionImpl perunSession, String message) throws InternalErrorException, WrongReferenceAttributeValueException, AttributeNotExistsException, WrongAttributeAssignmentException {
-		List<String> resolvingMessages = new ArrayList<>();
-		if(message == null) return resolvingMessages;
+	public List<AuditerMessage> resolveVirtualAttributeValueChange(PerunSessionImpl perunSession, AuditerMessage message) throws InternalErrorException, WrongReferenceAttributeValueException, AttributeNotExistsException, WrongAttributeAssignmentException {
+		List<AuditerMessage> resolvingMessages = new ArrayList<>();
+		if (message == null) return resolvingMessages;
 
-		Matcher banModificationMatcher = banModification.matcher(message);
-		List<Pair<Resource, Member>> listOfAffectedObjects = new ArrayList<>();
+		if (message.getEvent() instanceof BanSetForResource) {
+			return Collections.singletonList(resolveBanSetForResource(perunSession, (BanSetForResource) message.getEvent()));
+
+		} else if (message.getEvent() instanceof BanRemovedForResource) {
+			return Collections.singletonList(resolveBanRemovedForResource(perunSession, (BanRemovedForResource) message.getEvent()));
+
+		} else if (message.getEvent() instanceof BanUpdatedForResource) {
+			return Collections.singletonList(resolveBanUpdatedForResource(perunSession, (BanUpdatedForResource) message.getEvent()));
+
+		} else if (message.getEvent() instanceof BanSetForFacility) {
+			return resolveBanSetForFacility(perunSession, (BanSetForFacility) message.getEvent());
+
+		} else if (message.getEvent() instanceof BanRemovedForFacility) {
+			return resolveBanRemovedForFacility(perunSession, (BanRemovedForFacility) message.getEvent());
+
+		} else if (message.getEvent() instanceof BanUpdatedForFacility) {
+			return resolveBanUpdatedForFacility(perunSession, (BanUpdatedForFacility) message.getEvent());
+		}
+
+		return resolvingMessages;
+	}
+
+
+	private AuditerMessage resolveBanSetForResource(PerunSessionImpl perunSession, BanSetForResource banSetForResource) throws InternalErrorException, AttributeNotExistsException, WrongAttributeAssignmentException {
```

Comment:
Please, change the return type of these 3 methods to ```List<AuditerMessage>```. If no message needs to be generated, you can simply return empty List. Also, it is more consistent to have all of these 6 methods returning the same type.